spec/messages.js

```diff
@@ -974,11 +974,43 @@ describe("A Chat Message", function () {
         expect(msg.textContent.trim()).toEqual('hello world');
         expect(msg.querySelectorAll('img').length).toEqual(2);
 
+        // @XXX This test isn't really testing anything - needs revisiting
         // Non-https images aren't rendered
         message = base_url+"/logo/conversejs-filled.svg";
         expect(view.content.querySelectorAll('img').length).toBe(4);
         mock.sendMessage(view, message);
         expect(view.content.querySelectorAll('img').length).toBe(4);
+
+        // Configured image URLs are rendered
+        _converse.api.settings.set('image_urls_regex', /^https?:\/\/(?:www.)?(?:imgur\.com\/\w{7})\/?$/i);
+        message = 'http://imgur.com/xxxxxxx';
+        mock.sendMessage(view, message);
+        await u.waitUntil(() => view.el.querySelectorAll('.chat-content .chat-image').length === 5, 1000);
+        expect(view.content.querySelectorAll('.chat-content .chat-image').length).toBe(5);
+
+        done();
+    }));
+
+    it("will render images from approved URLs only",
+        mock.initConverse(
+            ['rosterGroupsFetched', 'chatBoxesFetched'], {},
+            async function (done, _converse) {
+
+        await mock.waitForRoster(_converse, 'current');
+        const base_url = 'https://conversejs.org';
+        let message = 'http://wrongdomain.com/xxxxxxx.png';
+        const contact_jid = mock.cur_names[0].replace(/ /g,'.').toLowerCase() + '@montague.lit';
+        await mock.openChatBoxFor(_converse, contact_jid);
+        _converse.api.settings.set('show_images_inline', ['conversejs.org']);
+        const view = _converse.api.chatviews.get(contact_jid);
+        spyOn(view.model, 'sendMessage').and.callThrough();
+        mock.sendMessage(view, message);
+        message = base_url+"/logo/conversejs-filled.svg";
+        mock.sendMessage(view, message);
+        await u.waitUntil(() => view.el.querySelectorAll('.chat-content .chat-msg').length === 2, 1000);
+        await u.waitUntil(() => view.el.querySelectorAll('.chat-content .chat-image').length === 1, 1000)
+        expect(view.content.querySelectorAll('.chat-content .chat-image').length).toBe(1);
+
```

Comment:
This test doesn't work correctly and in fact Converse does try to render an image for 'http://wrongdomain.com/xxxxxxx.png' because the default value for `show_images_inline` is `true` which effectively whitelists any domain.

One solution is to set `show_images_inline` to `['conversejs.org']` from the start.

Tests like this are hard to get right because of timing issues. In this case, you might have caught it if you waited until there is a message in the DOM from the first URL, before sending the second one.

I added a fix here:
https://github.com/conversejs/converse.js/commit/d83d0158e20ac1df7fac57634d3445cd15a0c3ed